chia/wallet/wallet_weight_proof_handler.py

```diff
@@ -37,13 +37,8 @@ def __init__(
             initializer=setproctitle,
             initargs=(f"{getproctitle()}_worker",),
         )
-        self._weight_proof_tasks: List[asyncio.Task[Tuple[bool, List[BlockRecord]]]] = []
 
     def cancel_weight_proof_tasks(self) -> None:
-        for task in self._weight_proof_tasks:
-            if not task.done():
-                task.cancel()
-        self._weight_proof_tasks = []
```

Comment:
Yes the weight proof validation runs as part of the new peak processing / long sync which gets canceled here https://github.com/Chia-Network/chia-blockchain/blob/5b3343b90e38c02c07d9a6387899c6d5e97f370e/chia/wallet/wallet_node.py#L407 so there is no need to pass down the cancelation to some additionally created tasks since the asyncio cancelation will take care of it yes.